ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobMetadata.java

```diff
@@ -38,6 +41,25 @@
   private int containerId;
   private long expirationTime;
   private long deletionTime;
+  private EncryptionOrigin encryptionOrigin;
+  private String vcrKmsContext;
+  private String cryptoAgentFactory;
+
+  /**
+   * Different encryption types for cloud stored blobs
+   */
+  public enum EncryptionOrigin {
+
+    /** No encryption */
+    NONE,
+    /** Encrypted by Router */
+    ROUTER,
+    /** Encrypted by VCR */
+    VCR,
+    /** Encrypted by Cloud Data Store */
+    CLOUD
```

Comment:
In prod, if we enable cloud encryption. How to notify a live VCR?

Reply:
CLOUD is here mainly for completeness, I'm not even sure when we would use it.

Reply:
Yes. I understand CLOUD is for completeness. 
I am thinking that if our VCRs are working in PROD and we are going to enable azure encryption. In this case, we need to shutdown all VCR and set `EncryptionOrigin` from NONE to CLOUD, and then enable azure encryption.

Reply:
Per discussion, removed CLOUD for now and made clear that encryptionOrigin is only from Ambry's perspective.